tests/acceptance/features/apiSharingNg/shareInvitations.feature

```diff
@@ -0,0 +1,99 @@
+Feature: Send a sharing invitations
+  As the owner of a resource
+  I want to be able to send invitations to other users
+  So that they can have access to it
+
+  https://owncloud.dev/libre-graph-api/#/drives.permissions/Invite
+
+  Background:
+    Given these users have been created with default attributes and without skeleton files:
+      | username |
+      | Alice    |
+      | Brian    |
+
+
+  Scenario Outline: send sharing invitation to user with different roles via the Graph API
+    Given user "Alice" has uploaded file with content "to share" to "/textfile1.txt"
+    And user "Alice" has created folder "FolderToShare"
+    When user "Alice" sends the following share invitation using the Graph API:
+      | resourceType | <resource-type> |
+      | resource     | <path>         |
+      | space        | Personal       |
+      | sharee       | Brian          |
+      | shareType    | user           |
+      | role         | <role>         |
```

Comment:
Maybe we should set the role as id in the test? and delete method `getRoleIdByName` ?

like:
```
When user "Alice" sends the following share invitation using the Graph API:
	      | space     | <space-name>                         |
	      | path      | <path>                               |
	      | shareWith | <sharee>                             |
              | role      | 2d00ce52-1fc2-4dbc-8b95-a73b73395f5a | # editor
```

or form the role as an array of permissions: 

<img width="421" alt="image" src="https://github.com/owncloud/ocis/assets/84779829/be7e451a-305a-4f2f-8ac7-4a9e8ba6a5bb">

```
When user "Alice" sends the following share invitation using the Graph API:
	      | space       | <space-name>                         |
	      | path        | <path>                               |
	      | shareWith   | <sharee>                             |
              | permissions |  permissions/create, path/read, quota/read, permissions/read, basic/read|
```

Reply:
> Maybe we should set the role as id in the test? and delete method `getRoleIdByName` ?
> 
> like:
> 
> ```
> When user "Alice" sends the following share invitation using the Graph API:
> 	      | space     | <space-name>                         |
> 	      | path      | <path>                               |
> 	      | shareWith | <sharee>                             |
>               | role      | 2d00ce52-1fc2-4dbc-8b95-a73b73395f5a | # editor
> ```
> 
> or form the role as an array of permissions:
> 
> <img alt="image" width="421" src="https://private-user-images.githubusercontent.com/84779829/291316848-be7e451a-305a-4f2f-8ac7-4a9e8ba6a5bb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI5NzQ5NTAsIm5iZiI6MTcwMjk3NDY1MCwicGF0aCI6Ii84NDc3OTgyOS8yOTEzMTY4NDgtYmU3ZTQ1MWEtMzA1YS00ZjJmLThhYzctNGE5ZThiYTZhNWJiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE5VDA4MzA1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTllMDVjNTI4YWQ3M2ZmZGVhZjI4MDk3ZTM5NzBhNDU3YjZlOGQyOTliMzZhNWU2YzdjNDBmZTMzNTA4ODI1ZjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.AugPDeGn2wpsG-b7aH1sMiJSUw_G7ZMw_hsUhkebBbE">
> ```
> When user "Alice" sends the following share invitation using the Graph API:
> 	      | space       | <space-name>                         |
> 	      | path        | <path>                               |
> 	      | shareWith   | <sharee>                             |
>               | permissions |  permissions/create, path/read, quota/read, permissions/read, basic/read|
> ```

Both ways seem unreadable, but if we have to do either way then I prefer to do with role id, the latter one seems tedious.

Reply:
or we can decide by `description` using resourceType.
`getRoleIdByName($baseUrl, $xRequestId, $user, $password, $role, $resouceType);`
if we don't want to remember id or make table long

Reply:
https://github.com/owncloud/ocis/blob/master/services/graph/pkg/unifiedrole/unifiedrole.go#L16-L58

we will use two options for creating sharing invitations, so our tests should be able to create invitations: 
- permissions.roles
- permissions.actions

so we need use table like: 
```
| shareWith | <sharee> | 
| role | fileEditor |
```
and: 
```
|permitions| permissions/create, path/read, quota/read .....|
```

I suggest to create one table in the GraphHelper where we will match all roles with id. and use roleNaming same as here https://github.com/owncloud/ocis/blob/master/services/graph/pkg/unifiedrole/unifiedrole.go#L16-L31
```
| fileEditor | 2d00ce52-1fc2-4dbc-8b95-a73b73395f5a|
| editor | fb6c3e19-e378-47e5-b277-9732f9de6e21 |
| spaceEditor | 58c63c02-1d89-4572-916a-870abc5a1b7d |
```

Reply:
looks like we have to make copy of https://github.com/owncloud/ocis/blob/master/services/graph/pkg/unifiedrole/unifiedrole.go#L16-L58 in the test and reuse them.
THanks Viktor, way to go with your suggestions

Reply:
refactored the method getRoleIdByName to get ids without API request.

Reply:
awesome. thank you)